pkgs/servers/sql/postgresql/ext/pgvecto-rs/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, postgresql
+, buildPgrxExtension
+, fetchFromGitHub
+, pkg-config
+, cargo-pgrx
+, fetchCrate
+, openssl
+, clang
+, nix-update-script
+}:
+
+let
+  cargo-pgrx_0_11_2 = cargo-pgrx.overrideAttrs (old: rec {
+    name = "cargo-pgrx-${version}";
+    version = "0.11.2";
+
+    src = fetchCrate {
+      pname = "cargo-pgrx";
+      inherit version;
+      hash = "sha256-8NlpMDFaltTIA8G4JioYm8LaPJ2RGKH5o6sd6lBHmmM=";
+    };
+
+    cargoDeps = old.cargoDeps.overrideAttrs (_: {
+      inherit src;
+      outputHash = "sha256-qTb3JV3u42EilaK2jP9oa5D09mkuHyRbGGRs9Rg4TzI=";
+    });
+  });
+
+in
+(buildPgrxExtension.override { cargo-pgrx = cargo-pgrx_0_11_2; }) rec {
+  inherit postgresql;
+
+  pname = "pgvecto-rs";
+  version = "0.2.0";
+
+  buildInputs = [ openssl ];
+  nativeBuildInputs = [ pkg-config clang ];
+
+  patches = [
+    # Tell the `c` crate to use the flags from the rust bindgen hook
+    ./0001-read-clang-flags-from-environment.diff
+    # Rust feature result_option_inspect is only stabilized on Rust 1.76.0,
+    # while nixpkgs is only on 1.75.0.
+    ./0002-enable-feature-result-option-inspect.diff
+  ];
+
+  src = fetchFromGitHub {
+    owner = "tensorchord";
+    repo = "pgvecto.rs";
+    rev = "v${version}";
+    hash = "sha256-30AzS9R01+Ntq/HKabn+3tRHiEe68pxoVK3sbr+wFWg=";
+  };
+
+  # package has git dependencies on Cargo.lock, so cargoHash does not work
```

Comment:
Not familiar with rust but what exactly does this mean?

Reply:
This means they have at least one dependency that is fetched from a git repo instead of from crates.io.
https://github.com/tensorchord/pgvecto.rs/blob/be5a816810f63cadada90a666cfc70db37c78a33/Cargo.toml#L35

I can try to clarify more in the comment and put there the link above as well. I took a look at the master branch now and that dependency is gone, so it seems this might stop being a problem in the future.